geopandas/plotting.py

```diff
@@ -340,16 +337,22 @@ def plot_series(s, cmap=None, color=None, ax=None, figsize=None, **style_kwds):
         style_kwds["vmin"] = style_kwds.get("vmin", values.min())
         style_kwds["vmax"] = style_kwds.get("vmax", values.max())
 
-    geom_types = s.geometry.type
+    # decompose GeometryCollections
+    geoms, multiindex = _flatten_multi_geoms(s.geometry, prefix="Geom")
+    values = np.take(values, multiindex, axis=0) if cmap else None
+    expl_series = gpd.GeoSeries(geoms)
```

Comment:
Actually, we don't really need it to be a Series, right? (GeometryArray should be fine enough?)

Reply:
We need a series as if there is no Multi geometry, original GeoSeries is returned. GeometryArray in that case raises error.

Reply:
I tried to make couple more changes to use GeometryArray, but it breaks the code at multiple places. I would keep GeoSeries there, but I am not aware how to import it without using `import geopandas` here.

Reply:
Yeah, I wanted to suggest to use GeometryArray always then, but eg inside the flatten method, we use `.str.startswith` which is a Series only method. 
Possible refactor for later then!